addons/im_livechat/static/src/core/thread_model_patch.js

```diff
@@ -34,6 +34,20 @@ patch(Thread.prototype, "im_livechat", {
         return this._super().filter((correspondent) => !correspondent.is_bot);
     },
 
+    get correspondent() {
+        let correspondent = this._super();
+        if (this.type === "livechat" && !correspondent) {
+            // Correspondent is only defined for 1-to-1 channels. But in the
+            // case of a livechat with an unknown user, the public user
+            // should be considered the correspondent even if multiple
+            // operators have been invited to the channel.
+            correspondent = this.channelMembers
+                .map((member) => member.persona)
+                .find((persona) => persona.is_public);
```

Comment:
Heuristics looks weak, I believe it's not working if the livechat is with a logged in user.

Maybe taking the lowest member id that is not the operator would work better?

Reply:
Good idea, updated :+1: